lib/semirel.gi

```diff
@@ -972,13 +972,14 @@ function(m)
   fi;
 
   #gens:=Set(GeneratorsOfMonoid(semi));
-  gens:=Set(Filtered(GeneratorsOfMonoid(semi), x-> not IsOne(x)));
+  one:=One(semi);
+  gens:=Set(Filtered(GeneratorsOfMonoid(semi), x -> x <> one));
```

Comment:
So, really this is the only changed line here, right? I'm not sure it's worth changing the other lines too (i.e. by replacing `One(semi)` by `one`).

Reply:
While I don't know what Wilf's motivation was, I think it's quite natural to use the already computed value of `one` instead of recomputing it a few times later on -- this is a (albeit fairly minor, but still natural) optimization.

Reply:
This ship has sailed but just to say that One is an attribute of a semigroup and so only calculated on the first call...